test/core.js

```diff
@@ -1,84 +1,122 @@
-
-QUnit.test( "jQuery( '#' )", function( assert ) {
-	assert.expect( 2 );
-
-	expectWarning( assert, "Selector, through the jQuery constructor, nothing but hash", function() {
-		var set = jQuery( "#" );
-		assert.equal( set.length, 0, "empty set" );
-	} );
-} );
-
-QUnit.test( "Attribute selectors with unquoted hashes", function( assert ) {
-	assert.expect( 31 );
+});
+
+QUnit.test("jQuery(html, props)", function (assert) {
+	assert.expect(3);
+
+	var $el = jQuery("<input/>", { name: "name", val: "value", size: 42 });
+
+	assert.equal($el.attr("name"), "name", "Name attribute");
+	assert.equal($el.attr("size"),
+		jQuery.isEmptyObject(jQuery.attrFn) ? undefined : "42", "Size attribute");
+	assert.equal($el.val(), "value", "Call setter method");
+});
+
+QUnit.test("jQuery( '#' )", function (assert) {
+	assert.expect(2);
+
+	expectWarning(assert, "Selector, through the jQuery constructor, nothing but hash", function () {
+		var set = jQuery("#");
+		assert.equal(set.length, 0, "empty set");
+	});
+});
+
+QUnit.test("Improperly closed elements", function (assert) {
+	assert.expect(4);
+
+	var oldWindowAlert = window.alert;
+	window.alert = function() {
+		assert.notOk("Called alert with "+ arguments)
+	};
+
+	expectWarning(assert, "Elements not self-closable nested wrong", 4, function () {
+		jQuery("<div><p/><span/></div>");
+		jQuery("<blockquote><p class='bad' /><span/></blockquote>");
+		jQuery("<div />").append("<div data-borked='y'><span/> <p/></div>");
+		jQuery("<script /><div class=afterscript/>")
+	});
+
+	expectNoWarning(assert, "Script doesn't run in processed strings", function () {
+		jQuery("<script>alert('YOU SHOULD NOT SEE THIS ALERT')</script>");
+		jQuery("<div>").append("<script src='YOU-SHOULD-NOT-SEE-THIS.SCRIPT'></script>");
+		jQuery("<script>alert('YOU SHOULD NOT SEE THIS CONSOLE')</script>");
+		// jQuery core code causes this to execute, and also it's async
+		// jQuery("<img src='not-found-this-is-ok.404' onerror=\"alert('YOU SHOULD NOT SEE THIS ONERROR')\" />");
+	});
+
+	expectNoWarning(assert, "Elements not self-closable but tolerable", function () {
+		jQuery("<div class=wonky />");
+		jQuery("<p style='width: 2%' />");
+		jQuery("<p />").append("<span aria-label='hello' />");
+	});
+
+	expectNoWarning(assert, "Bare elements", function () {
+		jQuery("<p/>").append("<strong>");
+		jQuery("<abbr />");
+		jQuery("<head />");
+	});
+
+	window.alert = oldWindowAlert;
+});
+
+QUnit.test("Attribute selectors with unquoted hashes", function (assert) {
+	assert.expect(31);
 
 	var markup = jQuery(
-			"<div>" +
-				"<div data-selector='a[href=#main]'></div>" +
-				"<a href='space#junk'>test</a>" +
-				"<link rel='good#stuff' />" +
-				"<p class='space #junk'>" +
-					"<a href='#some-anchor'>anchor2</a>" +
-					"<input value='[strange*=#stuff]' />" +
-					"<a href='#' data-id='#junk'>anchor</a>" +
-				"</p>" +
-			"</div>" ).appendTo( "#qunit-fixture" ),
+		"<div>" +
+		"<div data-selector='a[href=#main]'></div>" +
```

Comment:
The previous markup was more readable due to indentation. I'd keep the old styling or add indentation in the string itself, the new way is hard to read.

IMO it'd be good to revert all unrelated style changes from this PR, merge #339 & then only add new tests. We can always improve the styling in a separate PR later when we know ESLint actually checks it.